src/expr/src/scalar/func/impls/float64.rs

```diff
@@ -440,14 +440,20 @@ sqlfunc!(
         if !f.is_finite() {
             None
         } else {
-            let secs = f.trunc() as i64;
+            let mut secs = f.trunc() as i64;
```

Comment:
I'd like to see all of the `as` removed instead of rejiggering them, because it's still pretty unclear what's going on.

Reply:
I just removed all the integer to integer conversions using `as`. Unfortunately, as far as I'm aware there is no lossy conversion from `f64` to `i64` without using `as`.

Reply:
Indeed! Can you move this logic into ore then? From Nikhil:

> Over time, we should wrap up every as conversion into a descriptively named and documented method in mz_ore::cast (e.g., cast_usize_to_f64_lossy), and remove all the lint exceptions across the codebase.

Reply:
f64's range is greater than i64, and this will just saturate to the max i64 value in that case. Should we return an error in that case?

Reply:
Good point, I'll do that now.

Reply:
The desired behavior (error, saturate) might depend on each caller's needs. The cast fn might need to return an Option or enum to allow callers to decide what to do if it doesn't fit. ("not fit" probably means specific things for the fractional part, too, maybe that can be explicitly ignored by the cast fn, and force callers to care about it themselves?)

This specific sql function probably should error, especially because we don't support the infinite timestamps.

Reply:
Just updated to use the `ore` cast and return errors if we get out of the timestamp range.